llama-index-integrations/agent/llama-index-agent-openai/llama_index/agent/openai/base.py

```diff
@@ -54,6 +54,7 @@ def __init__(
         verbose: bool = False,
         max_function_calls: int = DEFAULT_MAX_FUNCTION_CALLS,
         default_tool_choice: str = "auto",
+        raise_error: bool = False,
```

Comment:
My general feedback is this arg is pretty generic, but its actual use only covers one very specific use-case

Is there any way we can think of to improve this? A better name? Handle more scenarios where errors are caught?

Reply:
Also, please double check the unit tests

Reply:
Hi Logan! Thank you for the comments! :)

So regarding this feature, absolutely! I understand there are multiple places where exceptions are raised which are captured by llama-index code.
This was motivated by Azure filtering error exception while streaming, I needed to capture in my side the exception. 

I noticed the functionality (i.e. the flag to re-reaise the exception) was there, although not end-to-end implemented.
My main objective was to introduce the parameter at an Agent level, as this was recommended in a thread in Discord.

I'm no expert of the whole llama-index codebase, but I am sure that with this small step, any other in need of this kind of feature will be able to adjust, in future PRs, the code to get exception in particular scenarios if they need!

Regarding the naming of the variable at an Agent level, wdyt of `capture_exception` or `rethrow_exception`, to be more explicit of not an error, but an exception.

And to finish, I checked the unit tests, they neither pass in the main branch (at least on my setup). I would take a look.

Please let me know about your thought on this! :sparkles: